File: credit/src/main/java/ru/patterns/credit/infrastructure/metric/MetricRegistry.java
package ru.patterns.credit.infrastructure.metric;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import ru.patterns.credit.infrastructure.messaging.sender.KafkaMetricSender;
import ru.patterns.credit.shared.kafka.MetricDto;

@Component
@RequiredArgsConstructor
public class MetricRegistry {
    private final KafkaMetricSender kafkaMetricSender;

    public void register(MetricDto dto) {
        kafkaMetricSender.send(dto);
    }
}

File: core/src/main/kotlin/ru/patterns/core/utils/metric/MetricRegistry.kt
package ru.patterns.core.utils.metric

import org.springframework.stereotype.Component
import ru.patterns.core.utils.metric.serialization.MetricDto

@Component
class MetricRegistry(
    private val kafkaMetricSender: KafkaMetricSender
) {
    fun register(metricDto: MetricDto) {
        kafkaMetricSender.sendObjectToKafkaAsync(metricDto)
    }
}